diff --git a/printrun/printrun_utils.py b/printrun/printrun_utils.py
index f6142bd..c8b8820 100644
--- a/printrun/printrun_utils.py
+++ b/printrun/printrun_utils.py
@@ -22,17 +22,22 @@ import subprocess
 import shlex
 import logging
 
+def install_path(sub):
+    import __main__ as main
+    return os.path.join(os.path.dirname(main.__file__), "..", sub)
+
 # Set up Internationalization using gettext
 # searching for installed locales on /usr/share; uses relative folder if not
 # found (windows)
 def install_locale(domain):
-    if os.path.exists('/usr/share/pronterface/locale'):
-        gettext.install(domain, '/usr/share/pronterface/locale', unicode = 1)
-    elif os.path.exists('/usr/local/share/pronterface/locale'):
-        gettext.install(domain, '/usr/local/share/pronterface/locale',
-                        unicode = 1)
-    else:
-        gettext.install(domain, './locale', unicode = 1)
+    candidates = [install_path('share/pronterface/locale'),
+                  '/usr/share/pronterface/locale',
+                  '/usr/local/share/pronterface/locale']
+    for path in candidates:
+        if os.path.exists(path):
+            gettext.install(domain, path, unicode = 1)
+            return
+    gettext.install(domain, './locale', unicode = 1)
 
 def setup_logging(out):
     logger = logging.getLogger()
@@ -49,7 +54,8 @@ def iconfile(filename):
         return pixmapfile(filename)
 
 def imagefile(filename):
-    for prefix in ['/usr/local/share/pronterface/images',
+    for prefix in [install_path('share/pronterface/images'),
+                   '/usr/local/share/pronterface/images',
                    '/usr/share/pronterface/images']:
         candidate = os.path.join(prefix, filename)
         if os.path.exists(candidate):
@@ -73,11 +79,13 @@ def lookup_file(filename, prefixes):
         return filename
 
 def pixmapfile(filename):
-    return lookup_file(filename, ['/usr/local/share/pixmaps',
+    return lookup_file(filename, [install_path('share/pixmaps'),
+                                  '/usr/local/share/pixmaps',
                                   '/usr/share/pixmaps'])
 
 def sharedfile(filename):
-    return lookup_file(filename, ['/usr/local/share/pronterface',
+    return lookup_file(filename, [install_path('share/pronterface'),
+                                  '/usr/local/share/pronterface',
                                   '/usr/share/pronterface'])
 
 def configfile(filename):
